src/shaders/fill_extrusion_pattern.fragment.glsl

```diff
@@ -38,7 +43,18 @@ void main() {
     vec4 out_color = mix(color1, color2, u_fade);
 
     out_color = out_color * v_lighting;
-
+#ifdef FAUX_AO
+    float intensity = u_ao[0];
+    float h = max(0.0, v_ao.z);
+    float h_floors = h / u_ao[1];
+    float y_shade = 1.0 - 0.9 * intensity * min(v_ao.y, 1.0);
+    float shade = (1.0 - 0.08 * intensity) * (y_shade + (1.0 - y_shade) * (1.0 - pow(1.0 - min(h_floors / 16.0, 1.0), 16.0))) + 0.08 * intensity * min(h_floors / 160.0, 1.0);
```

Comment:
I think it'd be helpful to have a bit more documentation/references/or the thought process that led to this shading equation.

The PR is really well described, but my concern is that this block feels like it may become difficult to change or tweak for someone else. It would take some time to guess and reason about how it was designed in order to make meaningful changes to it.

What could help is:
- extract some of these statements (including constants) as named variables, so that someone else reading that could tweak them
- document in code how this is designed, with the constraints/requirements (including what was mentioned in https://github.com/mapbox/mapbox-gl-js/pull/11926/files#r892771397)
- if any was used, have references to external reads 
- if possible, link to https://www.desmos.com/calculator to describe some part of how this behaves

Reply:
Thanks. Let me do that part tomorrow in this PR.